MLIR.cmake

```diff
@@ -77,6 +77,17 @@ function(add_onnx_mlir_dialect dialect)
   add_public_tablegen_target(OM${dialect}IncGen)
 endfunction()
 
+function(add_onnx_mlir_dialect_new tdfile dialect_namespace)
```

Comment:
Can we have a better name for this function? `_new` is not very descriptive.

I would recommend instead expanding the functionality of the original function to optionally take in the namespace and if the namespace exists, invoking the additional functionality.

Reply:
The namespace here means the name before the types or ops  in .mlir file, such as !onnx.StringType and onnx.Add. It is required for generating classes for Dialect and Types.
This function can replace the existing one after we change all the usages (for Krnl). How about change the old function into add_onnx_mlir_dialect_ops_only?

Reply:
My suggestion is to have a single function that optionally sets up the namespace if provided. That way the functionality will not change for any existing usage, but new usage that provides the namespace will make the additional invocations.

Reply:
> This function can replace the existing one after we change all the usages (for Krnl).

If it is really only temporary, then its ok. You may just add a TODO chentong near it, or call it `add_onnx_mlir_dialect_temporary_name`